### PR TITLE
detect/xbits: parse keywords w strtok_r 

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/4820

### CODE REVIEW

File: src/detect-xbits.c [hunk 1]
```diff
@@ -193,99 +188,109 @@ static int DetectXbitMatch (DetectEngineThreadCtx *det_ctx, Packet *p, const Sig
...
-            SCLogError(SC_ERR_PCRE_GET_SUBSTRING, "pcre_copy_substring failed");
-            return -1;
+    DetectXbitsData *cd = NULL;
+    char name[256] = "";
+    char copy[strlen(rawstr) + 1];
+    strlcpy(copy, rawstr, sizeof(copy));
+    char *context = NULL;
+    char *token = strtok_r(copy, ",", &context);
+    while (token != NULL) {
+        while (*token != '\0' && isblank(*token)) {
```
Comment (inashivb) [inline src/detect-xbits.c:206]: Could probably use https://github.com/OISF/suricata/pull/6847

File: doc/userguide/rules/xbits.rst [hunk 2]
```diff
@@ -21,7 +19,7 @@ Notes
    ``track ip_dst``, if you want to match on the server response,
    you check it (``isset``) with ``track ip_src``.
 
--  To not alert, use ``noalert;``
+-  To not alert, use ``noalert`` as a part of ``xbits`` options at the end.
```
Comment (inashivb) [inline doc/userguide/rules/xbits.rst:22]: Q: should it not be position specific?

**Comment by inashivb** [comment]
Needs some discussion as per input from Brandon so marking Draft for now.

**Comment by suricata-qa** [comment]
ERROR: QA failed on tlpw1_files_sha256.
  

Pipeline 5960

**Pull request closed without merging**